geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseConnectionPool.scala

```diff
@@ -146,47 +147,57 @@ object HBaseConnectionPool extends LazyLogging {
    * @param conf conf
    */
   def configureSecurity(conf: Configuration): Unit = synchronized {
-    if (User.isHBaseSecurityEnabled(conf)) {
-      val principal = conf.get(HBaseGeoMesaPrincipal)
-      val keytab = conf.get(HBaseGeoMesaKeyTab)
-
-      logger.debug(
-        s"Using Kerberos with principal '$principal', keytab '$keytab', " +
-            s"and Hadoop authentication method '${SecurityUtil.getAuthenticationMethod(conf)}'")
+    import AuthenticationTokenIdentifier.AUTH_TOKEN_TYPE
 
+    if (User.isHBaseSecurityEnabled(conf)) {
       val currentUser = UserGroupInformation.getCurrentUser
-
-      if (currentUser.hasKerberosCredentials && sameName(currentUser, principal)) {
-        logger.debug(s"User '$principal' is already authenticated")
+      if (currentUser.getCredentials.getAllTokens.asScala.exists(_.getKind == AUTH_TOKEN_TYPE)) {
+        logger.debug("Using existing HBase authentication token")
       } else {
-        if (currentUser.hasKerberosCredentials) {
-          logger.warn(
-            s"Changing global authenticated Hadoop user from '${currentUser.getUserName}' to '$principal' -" +
-                "this will affect any connections still using the old user")
+        val keytab = conf.get(HBaseGeoMesaKeyTab)
+        val rawPrincipal = conf.get(HBaseGeoMesaPrincipal)
+
+        if (keytab == null || rawPrincipal == null) {
+          lazy val missing =
+            Seq(HBaseGeoMesaKeyTab -> keytab, HBaseGeoMesaPrincipal -> rawPrincipal).collect { case (k, null) => k }
+          logger.warn(s"Security is enabled but missing credentials under '${missing.mkString("' and '")}'")
```

Comment:
Are the docs clear about setting both of these? 

And is it required?

Reply:
yeah the docs are clear: https://www.geomesa.org/documentation/stable/user/hbase/kerberos.html#kerberos
they are both required. if kerberos is enabled, you need a keytab for anything to work. previously this 
 block would fail with an NPE (or maybe just silently, i forget exactly), so this way you at least get a warning.